transforms/language/doc_quality/python/src/doc_quality_transform.py

```diff
@@ -0,0 +1,264 @@
+        table = TransformUtils.add_column(table=table, name="docq_contain_bad_word", content=docq_contain_bad_word)
+        table = TransformUtils.add_column(table=table, name="docq_bullet_point_ratio", content=docq_bullet_point_ratio)
+        table = TransformUtils.add_column(
+            table=table, name="docq_ellipsis_line_ratio", content=docq_ellipsis_line_ratio
+        )
+        table = TransformUtils.add_column(
+            table=table, name="docq_alphabet_word_ratio", content=docq_alphabet_word_ratio
+        )
+        table = TransformUtils.add_column(
+            table=table, name="docq_contain_common_en_words", content=docq_contain_common_en_words
+        )
+        table = TransformUtils.add_column(table=table, name="metakenlm_docq_perplex_score", content=docq_perplex_score)
+
+        if self.text_lang == "ja":
+            table = table.append_column("docq_avg_ja_sentence_len", pa.array(docq_avg_ja_sentence_len))
+            table = table.append_column("docq_first_ja_alphabet_pos", pa.array(docq_first_ja_alphabet_pos))
+
+        metadata = {
+            "total_docs_count": table.num_rows,
+        }
+
+        return [table], metadata
+
+
+class DocQualityTransformConfiguration(TransformConfiguration):
+
+    """
+    Provides support for configuring and using the associated Transform class include
+    configuration with CLI args.
+    """
+    def __init__(self):
+        super().__init__(
+            name=short_name,
+            transform_class=DocQualityTransform,
+            remove_from_metadata=[data_factory_internal_key],
+        )
+        self.daf = None
+
+    def add_input_params(self, parser: ArgumentParser) -> None:
+        """
+        Add Transform-specific arguments to the given  parser.
+        This will be included in a dictionary used to initialize the DocQualityTransform.
+        By convention a common prefix should be used for all transform-specific CLI args
+        (e.g, noop_, pii_, etc.)
+        """
+        parser.add_argument(
+            f"--{text_lang_cli_param}",
+            default="en",
+            help="language used in the text content"
+        )
+        parser.add_argument(
+            f"--{doc_content_column_cli_param}",
+            default="contents",
+            help="column name that contains document text",
+        )
+        parser.add_argument(
+            f"--{doc_id_column_cli_param}",
+            default="document_id",
+            help="column name that contains document id",
+        )
+        parser.add_argument(
+            f"--{bad_word_filepath_cli_param}",
+            type=str,
+            required=True,
+            help="path to bad word file: local folder (file or directory) that points to bad word file",
+        )
+        parser.add_argument(
+            f"--{kenLM_model_cli_param}",
+            type=str,
+            required=True,
+            help="path to kenLM model: local folder (file or directory) that points to kenLM model",
```

Comment:
Now that you use DAF, you are not restrictured to "local folder".  Need to change the help text on this and others.

Reply:
done